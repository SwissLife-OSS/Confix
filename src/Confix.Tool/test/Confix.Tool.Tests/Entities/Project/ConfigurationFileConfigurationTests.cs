using System.Text.Json.Nodes;
using Confix.Tool.Abstractions;

namespace ConfiX.Entities.Component.Configuration;

public class ConfigurationFileConfigurationTests : ParserTestBase
{
    [Fact]
    public void Parse_Should_BeValid()
    {
        ExpectValid(
            """
            {
                "type": "dotnet-appsettings",
                "additional": "property"
            }
        """);
    }

    [Fact]
    public void Parse_Should_BeValid_WhenEmpty()
    {
        ExpectValid(""" { } """);
    }

    [Fact]
    public void Parse_Should_BeValid_When_JustAString()
    {
        ExpectValid("\"inline\"");
    }

    public override object Parse(JsonNode json)
    {
        return ConfigurationFileConfiguration.Parse(json);
    }
}
