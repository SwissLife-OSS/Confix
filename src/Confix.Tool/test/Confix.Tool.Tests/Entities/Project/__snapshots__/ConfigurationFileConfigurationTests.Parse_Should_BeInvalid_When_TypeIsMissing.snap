{
  "Message": "Expected property 'type' in object",
  "Path": "$"
}
